include/cpid/cpid_macos.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdlib.h>
+#include <uuid/uuid.h>
+
+// Microsoft SAL annotations are not available on macOS
+// Define them to enable use for documentation purposes
+#ifndef _In_
+#define _In_
+#endif
+#ifndef _Out_
+#define _Out_
+#endif
+#ifndef _Inout_
+#define _Inout_
+#endif
+
+typedef struct cpid_struct *cpid_handle_t;
```

Comment:
I think `cpid_handle_t` should be an opaque `void*` as it is in the Windows implementation.